src/layouts/Header.tsx

```diff
@@ -10,52 +10,59 @@ import styles from './Header.less'
 
 const { Header } = Layout
 
-class HeaderView extends PureComponent {
-  state = {
-    visible: true
-  }
+function HeaderView (props) {
+  const [visible, setVisible] = useState(true)
+  // TO Verify: set logic for oldScrollTop and ticking
+  const [oldScrollTop, setOldScrollTop] = useState(document.body.scrollTop)
+  const [ticking, setTicking] = useState(true)
 
-  static getDerivedStateFromProps (props, state) {
-    if (!props.autoHideHeader && !state.visible) {
-      return {
-        visible: true
-      }
+  useEffect(() => {
+    // componentdidmount
+    document.addEventListener('scroll', handScroll, { passive: true })
+    // Specify how to clean up after this effect:
+    return function cleanup () {
+      // componentdidunmount
+      document.removeEventListener('scroll', handScroll)
     }
-    return null
-  }
+  })
 
-  componentDidMount () {
-    document.addEventListener('scroll', this.handScroll, { passive: true })
-  }
-
-  componentWillUnmount () {
-    document.removeEventListener('scroll', this.handScroll)
-  }
+  const { isMobile, handleMenuCollapse, setting } = props
+  const { navTheme, layout, fixedHeader } = setting
 
-  getHeadWidth = () => {
-    const { isMobile, collapsed, setting } = this.props
+  const getHeadWidth = () => {
+    const { isMobile, collapsed, setting } = props
```

Comment:
instead of unfolding the `props` variables in every function you can do it once, eg:
```
const { isMobile, collapsed, setting, dispatch } = props
```
And then use it in local functions.

- [ ] Consider unfolding props once